src/services/ConfirmStakeIntentService.ts

```diff
@@ -0,0 +1,156 @@
+import { interacts } from '@openst/mosaic-contracts';
+import { EIP20CoGateway } from '@openst/mosaic-contracts/dist/interacts/EIP20CoGateway';
+import Observer from '../observer/Observer';
+import { MessageRepository } from '../repositories/MessageRepository';
+import Gateway from '../models/Gateway';
+import Message from '../models/Message';
+import Utils from '../Utils';
+import { AUXILIARY_GAS_PRICE } from '../Constants';
+
+import StakeRequestRepository from '../repositories/StakeRequestRepository';
+import Logger from '../Logger';
+
+const Mosaic = require('@openst/mosaic.js');
+
+const { ProofGenerator } = Mosaic.Utils;
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+/**
+ * Class collects all non confirmed pending messages and confirms those messages.
+ */
+export default class ConfirmStakeIntentService extends Observer<Gateway> {
+  private messageRepository: MessageRepository;
+
+  private stakeRequestRepository: StakeRequestRepository;
+
+  private originWeb3: object;
+
+  private auxiliaryWeb3: object;
+
+  private gatewayAddress: string;
+
+  private coGatewayAddress: string;
+
+  private auxiliaryWorkerAddress: string;
+
+  /**
+   * Constructor of class ConfirmStakeIntentService;
+   *
+   * @param messageRepository Instance of message repository.
+   * @param stakeRequestRepository Instance of stake request repository.
+   * @param originWeb3 Instance of origin chain web3.
+   * @param auxiliaryWeb3 Instance of auxiliary chain web3.
+   * @param gatewayAddress Origin chain gateway address.
+   * @param coGatewayAddress Auxiliary chain gateway address.
+   * @param auxiliaryWorkerAddress Auxiliary chain werker address.
+   */
+  public constructor(
+    messageRepository: MessageRepository,
+    stakeRequestRepository: StakeRequestRepository,
+    originWeb3: object,
+    auxiliaryWeb3: object,
+    gatewayAddress: string,
+    coGatewayAddress: string,
+    auxiliaryWorkerAddress: string,
+  ) {
+    super();
+
+    this.messageRepository = messageRepository;
+    this.stakeRequestRepository = stakeRequestRepository;
+    this.originWeb3 = originWeb3;
+    this.auxiliaryWeb3 = auxiliaryWeb3;
+    this.gatewayAddress = gatewayAddress;
+    this.coGatewayAddress = coGatewayAddress;
+    this.auxiliaryWorkerAddress = auxiliaryWorkerAddress;
+  }
+
+  /**
+   * This method reacts on changes when GatewayProven entity is received.
+   *
+   * Gateway model first object is selected because of update argument interface. From
+   * ProveGatewayHandler single Gateway model is passed.
+   *
+   * Messages to be send for confirmation is fetched and confirmStakeIntent is called.
+   *
+   * @param gateway List of Gateway models.
+   */
+  public async update(gateway: Gateway[]): Promise<void> {
+    const provenGateway: Gateway = gateway[0];
```

Comment:
It should assert for zero and more than 1 entry

Reply:
Done!